src/twisted/python/test/test_textattributes.py

```diff
@@ -5,23 +5,829 @@
 Tests for L{twisted.python.textattributes}.
 """
 
+import functools
+import operator
+
 from twisted.trial import unittest
-from twisted.python._textattributes import DefaultFormattingState
+from twisted.python._textattributes import (_Attribute,
+                                            _BackgroundColorAttr,
+                                            CharacterAttributesMixin,
+                                            _ColorAttr,
+                                            _ColorAttribute,
+                                            DefaultFormattingState,
+                                            flatten,
+                                            _ForegroundColorAttr,
+                                            _FormattingStateMixin,
+                                            _OtherAttr,
+                                            _NormalAttr)
+
+
+class RecordsFakeAttribute(object):
+    """
+    Records calls to a L{FakeAttributes} instance.
+
+    @ivar serializeCalls: The arguments passed to each call to the
+        containing object's C{serialize} method
+    @type serializeCalls: a L{list} of L{tuple}s, each of which are
+        the arguments for one call to C{serialize}
+    """
+
+    def __init__(self):
+        self.serializeCalls = []
+
+
+
+class FakeAttribute(_Attribute):
```

Comment:
Why note test a real _Attribute, and give it a write() that appends to a list or something?